### PR TITLE
Logstash adds update strategy

### PR DESCRIPTION
**Pull request by kaisecheng**
Logstash adds `updateStrategy` in CRD which is a [StatefulSetUpdateStrategy](https://pkg.go.dev/k8s.io/api/apps/v1#StatefulSetUpdateStrategy)
The default type is "RollingUpdate"

sample resource

```yaml
apiVersion: logstash.k8s.elastic.co/v1alpha1
kind: Logstash
metadata:
  name: logstash-sample
spec:
  count: 3
  version: 8.11.3
  pipelines:
    - pipeline.id: main
      pipeline.workers: 2
      config.string: |
        input { exec { command => 'uptime' interval => 10 } } 
        output { 
          stdout {}
        }
  updateStrategy: 
    type: "RollingUpdate"
    rollingUpdate:
      partition: 1
      maxUnavailable: 1
```

### CODE REVIEW

**Review by robbavey — APPROVED** [review]
LGTM!

**Comment by kaisecheng** [comment]
@pebrc @thbkrkr @barkbay This is ready for review

File: pkg/controller/logstash/logstash_controller_test.go [hunk 1]
```diff
@@ -201,6 +210,14 @@ func TestReconcileLogstash_Reconcile(t *testing.T) {
...
 					Version: "8.6.1",
 					Count:   1,
+					UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+						Type: appsv1.RollingUpdateStatefulSetStrategyType,
+						RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+							MaxUnavailable: &intstr.IntOrString{
+								IntVal: 1,
+							},
+						},
+					},
```
Comment (pebrc) [inline pkg/controller/logstash/logstash_controller_test.go:220]: Not from this PR but I am a bit confused by this test. Not sure what it actually verifies. It seems it it is just retrieving the Logstash resource again that we just put in as an expected object.

**Review by pebrc — APPROVED** [review]

File: pkg/controller/logstash/logstash_controller_test.go (hunk 1 above)
Comment (kaisecheng) [reply pkg/controller/logstash/logstash_controller_test.go:220]: It verifies the value of LogstashStatus is up-to-date

Comment (pebrc) [reply pkg/controller/logstash/logstash_controller_test.go:220]: I see. But for the addition of the `UpdateStrategy` a test would be more useful that verifies that the `UpdateStrategy` specified in the Logstash resource is actually applied to the `StatefulSet` that is being created.

Comment (kaisecheng) [reply pkg/controller/logstash/logstash_controller_test.go]: test is updated

**Review by pebrc — APPROVED** [review]

**Pull request merged**